File: AnimacoesSimples/app/src/main/java/com/caiodetore/animacoessimples/MainActivity.java
package com.caiodetore.animacoessimples;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.LinearLayout;

import com.bhargavms.dotloader.DotLoader;

public class MainActivity extends AppCompatActivity {
    private LinearLayout layoutDotLoader;
    private DotLoader dotLoader;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dotLoader = findViewById(R.id.text_dot_loader);
//        layoutDotLoader = findViewById(R.id.layout_dot_loader);

        dotLoader.postDelayed(() -> dotLoader.setNumberOfDots(3), 100);

    }
}

File: DesignTreino/app/src/main/java/com/caiodetore/designtreino/MainActivity.kt
package com.caiodetore.designtreino

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Switch
import androidx.appcompat.app.AppCompatDelegate
import androidx.appcompat.app.AppCompatDelegate.MODE_NIGHT_NO
import androidx.appcompat.app.AppCompatDelegate.MODE_NIGHT_YES
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val outlineText = findViewById<TextInputLayout>(R.id.textInputLayout)
        val textInputEdit = findViewById<TextInputEditText>(R.id.textInputEditText)
        val button = findViewById<Button>(R.id.button)
        val switch = findViewById<Switch>(R.id.switch1)
        val texto = textInputEdit.text

        switch.setOnClickListener {
            if (switch.isChecked) AppCompatDelegate.setDefaultNightMode(MODE_NIGHT_YES)
            else AppCompatDelegate.setDefaultNightMode(MODE_NIGHT_NO)
        }

        button.setOnClickListener{
            outlineText.isHelperTextEnabled = true
            outlineText.helperText = texto.toString()
        }
    }
}